simpleclient_pushgateway/src/main/java/io/prometheus/client/exporter/PushGateway.java

```diff
@@ -268,9 +268,10 @@ void doRequest(CollectorRegistry registry, String job, Map<String, String> group
         writer.close();
       }
 
-      int response = connection.getResponseCode();
-      if (response != HttpURLConnection.HTTP_ACCEPTED) {
-        throw new IOException("Response code from " + url + " was " + response);
+      int responseCode = connection.getResponseCode();
+      if (responseCode != HttpURLConnection.HTTP_ACCEPTED) {
+        throw new IOException("Response code from " + url + " was " + responseCode
+            + ", message: " + connection.getResponseMessage());
```

Comment:
Yeah, found the relevant code: http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/7u40-b43/sun/net/www/protocol/http/HttpURLConnection.java#1622

I'm afraid it doesn't look like this is going to be possible.